.github/workflows/test-cartesian.yml

```diff
@@ -20,35 +20,36 @@ concurrency:
 
 jobs:
   test-cartesian:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
+        codegen-factor: [internal, dace]
+        os: ["ubuntu-latest"]
         python-version: ["3.10", "3.11"]
-        tox-factor: [internal, dace]
+      fail-fast: false
```

Comment:
What is it doing in this context and what it the reason? maybe document.

Reply:
`fail-fast: true`, which is the default, makes the action to stop all matrix runs immediately after the first failure, which is a sensible default to save CI minutes (for [reference](https://www.edwardthomson.com/blog/github_actions_6_fail_fast_matrix_workflows)). 

In this particular case, I added the `fail: fast` here just is also present in the other subpackages, although I can't remember why we set it up like this. Probably because since gt4py is open source and has plenty of free CI minutes, the cost is not a critical factor and it might be still useful to see if all combinations of parameters in the matrix fail or only some of them. But we could also remove this settings from all the workflows if it makes more sense....